public/app/features/dashboard-scene/scene/DashboardSceneUrlSync.ts

```diff
@@ -17,41 +17,41 @@ export class DashboardSceneUrlSync implements SceneObjectUrlSyncHandler {
 
   getUrlState(): SceneObjectUrlValues {
     const state = this._scene.state;
-    return { inspect: state.inspectPanelId, viewPanel: state.viewPanelId };
+    return { inspect: state.inspectPanelKey, viewPanel: state.viewPanelKey };
   }
 
   updateFromUrl(values: SceneObjectUrlValues): void {
-    const { inspectPanelId, viewPanelId } = this._scene.state;
+    const { inspectPanelKey: inspectPanelId, viewPanelKey: viewPanelId } = this._scene.state;
     const update: Partial<DashboardSceneState> = {};
 
     // Handle inspect object state
     if (typeof values.inspect === 'string') {
-      const panel = findVizPanelById(this._scene, values.inspect);
+      const panel = findVizPanelByKey(this._scene, values.inspect);
```

Comment:
I reverted back to using keys here, as it becomes a lot easier to refer to the repeated panels using string keys than new numeric ids. 

We still support old urls. So `?inspect=2` works and is the same as `?inspect=panel-2`